src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php

```diff
@@ -659,6 +659,19 @@ public function testWebLink()
         $this->assertTrue($container->hasDefinition('web_link.add_link_header_listener'));
     }
 
+    public function testMessengerServicesRemovedWhenDisabled()
+    {
+        $container = $this->createContainerFromFile('full');
+        $this->assertFalse($container->has('console.command.messenger_consume_messages'));
+        $this->assertFalse($container->has('console.command.messenger_debug'));
+        $this->assertFalse($container->has('console.command.messenger_stop_workers'));
+        $this->assertFalse($container->has('console.command.messenger_setup_transports'));
+        $this->assertFalse($container->has('console.command.messenger_failed_messages_retry'));
+        $this->assertFalse($container->has('console.command.messenger_failed_messages_show'));
+        $this->assertFalse($container->has('console.command.messenger_failed_messages_remove'));
+        $this->assertFalse($container->has('cache.messenger.restart_workers_signal'));
```

Comment:
Maybe adding `assertTrue` to the following testcase would be useful? (`testMessenger()`)

Reply:
Added 

`$this->assertTrue($container->hasDefinition('console.command.messenger_consume_messages'));`

Should be enough since the cache is a dependency of that service?